test-proxy/src/main/java/com/google/cloud/bigtable/testproxy/CbtTestProxy.java

```diff
@@ -0,0 +1,768 @@
+      Duration newTimeout, BigtableDataSettings.Builder settingsBuilder) {
+    // TODO(developer): remove the initialRpcTimeout update below by updating the client library.
+    Duration initialRpcTimeout =
+        settingsBuilder
+            .stubSettings()
+            .bulkMutateRowsSettings()
+            .getRetrySettings()
+            .getInitialRpcTimeout();
+    if (initialRpcTimeout.compareTo(newTimeout) > 0) {
+      // Total timeout is smaller than initialRpcTimeout, which will cause deadline-related problem.
+      initialRpcTimeout = newTimeout;
+    }
+    settingsBuilder
+        .stubSettings()
+        .bulkMutateRowsSettings()
+        .retrySettings()
+        .setTotalTimeout(newTimeout)
+        .setInitialRpcTimeout(initialRpcTimeout);
+
+    settingsBuilder.stubSettings().mutateRowSettings().retrySettings().setTotalTimeout(newTimeout);
+
+    settingsBuilder.stubSettings().readRowSettings().retrySettings().setTotalTimeout(newTimeout);
+
+    settingsBuilder.stubSettings().readRowsSettings().retrySettings().setTotalTimeout(newTimeout);
+
+    settingsBuilder
+        .stubSettings()
+        .sampleRowKeysSettings()
+        .retrySettings()
+        .setTotalTimeout(newTimeout);
+
+    settingsBuilder
+        .stubSettings()
+        .checkAndMutateRowSettings()
+        .retrySettings()
+        .setTotalTimeout(newTimeout);
+
+    settingsBuilder
+        .stubSettings()
+        .readModifyWriteRowSettings()
+        .retrySettings()
+        .setTotalTimeout(newTimeout);
+
+    return settingsBuilder;
+  }
+
+  /** Helper method to get a client object by its id. */
+  private CbtClient getClient(String id) throws StatusException {
+    CbtClient client = idClientMap.get(id);
+    if (client == null) {
+      throw Status.NOT_FOUND.withDescription("Client " + id + " not found.").asException();
+    }
+    return client;
+  }
+
+  @Override
+  public synchronized void createClient(
+      CreateClientRequest request, StreamObserver<CreateClientResponse> responseObserver) {
+    if (request.getClientId().isEmpty()
+        || request.getProjectId().isEmpty()
+        || request.getInstanceId().isEmpty()
+        || request.getDataTarget().isEmpty()) {
+      responseObserver.onError(
+          Status.INVALID_ARGUMENT
+              .withDescription("clientId, projectId, instanceId, and dataTarget must be provided.")
+              .asException());
+      return;
+    }
+
+    if (idClientMap.get(request.getClientId()) != null) {
+      responseObserver.onError(
+          Status.ALREADY_EXISTS
+              .withDescription("Client " + request.getClientId() + " already exists.")
+              .asException());
+      return;
+    }
+
+    BigtableDataSettings.Builder settingsBuilder = BigtableDataSettings.newBuilder();
+    if (request.hasPerOperationTimeout()) {
+      com.google.protobuf.Duration timeoutFromReq = request.getPerOperationTimeout();
```

Comment:
Can you simplify this to Duration newTimeout = Duration.fromMillis(Durations.toMillis(request.getPerOperationTimeout());